packages/binding-coap/src/coap-server.ts

```diff
@@ -435,6 +435,12 @@ export default class CoapServer implements ProtocolServer {
     }
 
     private async handleRequest(req: IncomingMessage, res: OutgoingMessage) {
+        if (req.rsinfo.port === 0) {
```

Comment:
since `port` is of type `number` ... does it make sense to check for negative values also?
```suggestion
        if (req.rsinfo.port <= 0) {
```

Reply:
Good point! Maybe we could even have a check for the full port range, to also reflect the original error message.

```suggestion
        const sourcePort = req.rsinfo.port;
        const hasInvalidPortRange =  sourcePort < 1 || sourcePort > 65535;
        if (hasInvalidPortRange) {
```

Reply:
Took the freedom of already committing this in a640d6cfb8247f386d89d86e6a5f0fe8fb13625b, hope the change is fine for you as well 😅